src/transformations/correction/valid_point.jl

```diff
@@ -0,0 +1,24 @@
+#=
+# Valid Points
+
+We define a point as valid if it has no NaN or Inf values.
+
+Other corrections are the domain of the user and can be defined relatively easily using this template.
+
+=#
+
+struct ValidPoint <: GeometryCorrection end
+
+application_level(::ValidPoint) = Union{GI.LineStringTrait, GI.LinearRingTrait}
+
+function (::ValidPoint)(::Union{GI.LineStringTrait, GI.LinearRingTrait}, geom)
+    new_coords = NTuple{2, Float64}[]
+    sizehint!(new_coords, GI.npoint(geom))
+    for coord in GI.getpoint(geom)
+        x, y = GI.x(coord), GI.y(coord)
+        if isfinite(x) && isfinite(y)
+            push!(new_coords, (x, y))
+        end
```

Comment:
Is it better to iteratively delete elements from the array, rather than going through and reassembling?  Not sure here.  Will have to benchmark it...

Reply:
Delete is pretty slow

Reply:
It's not guaranteed that the number of invalid points will be less than the number of valid points - consider a projected polygon outside of the current projection, for example...

Reply:
Ok delete will be slow then. Maybe generating a bitindex is the fastest? I guess a benchmark is the only way to know

Reply:
I want to improve this but we can always benchmark later.